instant-xml/src/impls.rs

```diff
@@ -20,8 +24,165 @@ impl<'xml> FromXml<'xml> for bool {
 
     fn deserialize(deserializer: &mut Deserializer) -> Result<Self, Error> {
         match deserializer.consume_next_type() {
-            EntityType::Element => deserializer.deserialize_bool(BoolVisitor),
+            EntityType::Element => deserializer.deserialize_element(BoolVisitor),
             EntityType::Attribute => deserializer.deserialize_attribute(BoolVisitor),
         }
     }
 }
+
+struct NumberVisitor<T>
```

Comment:
Similar to what I mentioned on the serializer side, I think we should have a helper type here that can handle all of these conversions which can basically rely on `FromStr`.

Reply:
done